### DISCUSSION TITLE
Unable to launch Chromium/chrome using Jenkins Build

### DISCUSSION THREAD

**Discussion by dhanushng88** [getgauge/taiko#2092] [General]
Hello Team,

Can anyone help me out in identifying what's the issue with launching chromium using jenkins
Belo are the console errors we are facing[This was working as excepted from many months] but suddenly from last week we are seeing this issue.
Failed to capture screenshot on failure.
Error: Browser or page not initialized. Call `openBrowser()` before using this API
[runner: 4]     Error Message: Error: Failed to launch chrome!
    [17195:17195:0531/075543.616525:ERROR:browser_main_loop.cc(1386)] Unable to open X display.
    
    Stacktrace: 
    Error: Failed to launch chrome!
    [17195:17195:0531/075543.616525:ERROR:browser_main_loop.cc(1386)] Unable to open X display.
    
        at onClose (node_modules/taiko/lib/browserFetcher.js:282:11)
        at ChildProcess.helper.addEventListener (node_modules/taiko/lib/browserFetcher.js:268:11)
        at ChildProcess.emit (events.js:198:13)

**Taiko version :** 1.0.3
Gauge version: 1.0.7
Commit Hash: ed7b4fd6

Plugins
-------
html-report (4.0.10) 
js (2.3.8)
json-report (0.3.3)  
reportportal (1.6.0) 
screenshot (0.0.1)   
xml-report (0.2.3)  

**In browser.js file we are using this code**
beforeScenario(async () => {
    if (process.env.api_test_run === 'false') {
        await setConfig({ navigationTimeout: parseInt(process.env.test_timeout) });

        if (process.env.host === "local") {
            await openBrowser({ headless: false, ignoreCertificateErrors: true, args: ['--no-sandbox', '--start-fullscreen', '--disable-web-security','--disable-RendererCodeIntegrity'] })
            await deleteCookies();
        } else if (process.env.host === "docker") {
            await openBrowser({ headless: true, ignoreCertificateErrors: true, args: ['--no-sandbox', '--start-fullscreen', '--disable-web-security'] })
            await deleteCookies();
        }
    }
});

afterScenario(async () => {
    if (process.env.api_test_run === 'false') {
        await closeBrowser();
    }
    
});

**Comment by zabil** [comment]
```
Unable to open X display.
```

This error happens when chromium is launched on headful mode on a container like docker. Maybe the environment variables values are not set correctly causing it to launch with `headless: false` option.

**Reply by dhanushng88** [reply]
@zabil Can you suggest me how to fix this?
i am using the below code in browser.js file
 if (process.env.host === "local") {
        await openBrowser({ headless: false, ignoreCertificateErrors: true, args: ['--no-sandbox', '--start-fullscreen', '--disable-web-security','--disable-RendererCodeIntegrity'] })
        await deleteCookies();
    } else if (process.env.host === "docker") {
        await openBrowser({ headless: true, ignoreCertificateErrors: true, args: ['--no-sandbox', '--start-fullscreen', '--disable-web-security'] })
        await deleteCookies();
    }
}

**Comment by zabil** [comment]
How are you setting the `host` environment variable.

**Reply by dhanushng88** [reply]
In env folder--> default.properties file

# Script Run Mode (local, docker)
host = local

**Reply by zabil** [reply]
Please check that you have the right property and environment loaded when running Gauge
You can verify this by printing the value of `process.env.host` and checking if it is picking up the right value.

There is more documentation here about how to set and use environments

https://docs.gauge.org/configuration.html?os=windows&language=javascript&ide=vscode#using-environments-in-a-gauge-project

**Reply by dhanushng88** [reply]
@zabil First of thanks for your replies and suggestions..
Yes we have followed the same on default-->common.properties as specified in the gauge document
But this was working as expected before and we haven't changed anything in the above files.
only from last week Friday onwards we are seeing this on Jenkins run, but works fine in our local run

Is this something relates to **Fix update chromium workflow #2083** which is looked by @NivedhaSenthil  which is causing this issue?

**Reply by zabil** [reply]
That strange, very unlikely to be related to #2083 you could try modifying your build scripts or docker file to verify this.

**Reply by dhanushng88** [reply]
oh ok.. let me check once 
![image](https://user-images.githubusercontent.com/52316009/120320852-4762b680-c300-11eb-8e6d-c426a7cb5021.png)
